pkg/api/norman/customization/cluster/validator_test.go

```diff
@@ -1,193 +0,0 @@
-                "failuresOnly": false,
-                "overrideBenchmarkVersion": "rke-cis-1.4",
-                "profile": "permissive",
-                "skip": null
-            }
-        },
-        "scheduleConfig": {
-            "cronSchedule": ""
-        }
-    },
-    "type": "cluster",
-    "windowsPreferedCluster": false
-}
-`
-
-const clusterTemplateSpecJSON = `
-{
-    "dockerRootDir": "/var/lib/docker",
-    "enableClusterAlerting": false,
-    "enableClusterMonitoring": false,
-    "enableNetworkPolicy": false,
-    "labels": {},
-    "clusterTemplateRevisionId": "cattle-global-data:ctr-xxxxx",
-    "name": "testclusterfromtemplate",
-    "rancherKubernetesEngineConfig": {
-    },
-    "scheduledClusterScan": {
-        "enabled": true,
-        "scanConfig": {
-            "cisScanConfig": {
-                "failuresOnly": false,
-                "overrideBenchmarkVersion": "rke-cis-1.4",
-                "profile": "permissive",
-                "skip": null
-            }
-        },
-        "scheduleConfig": {
-            "cronSchedule": ""
-        }
-    },
-    "type": "cluster"
-}
-`
-
-func TestValidateScheduledClusterScan(t *testing.T) {
-	var clusterSpec mgmtclient.Cluster
-	err := json.Unmarshal([]byte(clusterSpecJSON), &clusterSpec)
-	if err != nil {
-		logrus.Errorf("error unmarshaling clusterspec: %v", err)
-		t.FailNow()
-	}
-
-	clusterSpec.ScheduledClusterScan.ScheduleConfig.Retention = -1
-	err = validateScheduledClusterScan(&clusterSpec)
-	if err == nil {
-		logrus.Errorf("expected error")
-		t.FailNow()
-	}
-	clusterSpec.ScheduledClusterScan.ScheduleConfig.Retention = 3
-
-	clusterSpec.ScheduledClusterScan.ScheduleConfig.CronSchedule = "junk"
-	err = validateScheduledClusterScan(&clusterSpec)
-	if err == nil {
-		logrus.Errorf("expected error")
-		t.FailNow()
-	}
-
-	everyHour := "0 */1 * * *"
-	clusterSpec.ScheduledClusterScan.ScheduleConfig.CronSchedule = everyHour
-	err = validateScheduledClusterScan(&clusterSpec)
-	if err != nil {
-		logrus.Errorf("not expecting error, got: %v", err)
-		t.FailNow()
-	}
-
-	everyMinute := "* * * * *"
-	clusterSpec.ScheduledClusterScan.ScheduleConfig.CronSchedule = everyMinute
-	err = validateScheduledClusterScan(&clusterSpec)
-	if err == nil {
-		logrus.Errorf("expected error")
-		t.FailNow()
-	}
-
-	clusterSpec.ScheduledClusterScan.ScanConfig.CisScanConfig.DebugMaster = true
-	err = validateScheduledClusterScan(&clusterSpec)
-	if err != nil {
-		logrus.Errorf("not expecting error, got: %v", err)
-		t.FailNow()
-	}
-}
-
-func TestClusterTemplateValidateScheduledClusterScan(t *testing.T) {
-	var clusterTemplateSpec mgmtclient.Cluster
-	err := json.Unmarshal([]byte(clusterTemplateSpecJSON), &clusterTemplateSpec)
-	if err != nil {
-		logrus.Errorf("error unmarshaling clusterspec: %v", err)
-		t.FailNow()
-	}
-
-	err = validateScheduledClusterScan(&clusterTemplateSpec)
-	if err != nil {
-		logrus.Errorf("not expecting error, got: %v", err)
-		t.FailNow()
-	}
-}
```

Comment:
We should add to integration test to make sure that creating a cluster using a clusterTemplateSpec where clusterSpec.ScheduledClusterScan.ScheduleConfig.CronSchedule deploys correctly and does not error out

Reply:
This would be a good place to add test with a clustertemplate having scheduledScan defined in it https://github.com/rancher/rancher/blob/f98b906db2fb8516e06560f9455af0c5f5859003/tests/integration/suite/test_clustertemplate.py
Refer test on https://github.com/rancher/rancher/blob/f98b906db2fb8516e06560f9455af0c5f5859003/tests/integration/suite/test_clustertemplate.py#L64 which shows how to test cluster creation using a template